routes/apps.js

```
@@ -0,0 +1,107 @@
+const express = require('express');
+const debug = require('debug')('sc2:server');
+const crypto = require('crypto');
+const _ = require('lodash');
+const { authenticate } = require('../helpers/middleware');
+const { sleep } = require('../helpers/utils');
+const config = require('../config.json');
+const router = express.Router();
+
+/** Get applications */
+router.get('/', async (req, res, next) => {
+  const apps = await req.db.apps.findAll({ attributes: { exclude: ['secret'] } });
+  res.json(apps);
+});
+
+/** Get my applications */
+router.all('/me', authenticate('user'), async (req, res, next) => {
+  const apps = await req.db.apps.findAll({ where: { owner: req.user } });
+  res.json(apps);
+});
+
+/** Get application details */
+router.get('/@:clientId', async (req, res, next) => {
+  const { clientId } = req.params;
+  const app = await req.db.apps.findOne({ where: { client_id: clientId } });
+  if (!app) return next();
+  if (!req.user || app.owner !== req.user) {
+    app.secret = undefined;
+  }
+  res.json(app);
+});
+
+/** Create application */
+router.post('/@:clientId', authenticate('user'), async (req, res, next) => {
+  const { clientId } = req.params;
+
+  /** Wait 5 seconds to insure the newly created account is available on the node */
+  await sleep(5000);
```

Should not delay API. Would be better to delay on front-end `CreateApp.js:75`
It should throw an error if the account is not available with msg indicating account creation can take upto 5 sec. That way developer would add sleep in front-end if he missed it.